src/eval.c

```diff
@@ -3605,6 +3617,296 @@ get_lit_string_tv(char_u **arg, typval_T *rettv, int evaluate)
     return OK;
 }
 
+    static int
+get_repeated_single_quotes_length(char_u *x, char_u *whole_start)
+{
+    char_u *follower;
+    char_u *following;
+
+    if (*x != '\'')
+    {
+        return 0;
+    }
+
+    for (follower = x; *follower == '\''; --follower)
```

Comment:
If there is a multibyte encoding which trailing byte can be `'`, this might not work well.
I'm not sure whether such strange encoding really exists. (At least, this is safe for UTF-8, Shift_JIS and EUC-JP.)

Reply:
I think so too.
mmm... Is a counterplan for this needed?

Reply:
@k-takata I removed this algorithm from the implementation!
Please see new implementation :pray: